extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/QuarkusErrorHandler.java

```diff
@@ -118,8 +122,7 @@ public void accept(Throwable throwable) {
             event.response().end();
             return;
         }
-        String accept = event.request().getHeader("Accept");
-        if (accept != null && accept.contains("application/json")) {
+        if (produceJson(event.request())) {
```

Comment:
This means we will produce JSON even if the client used `Accept: text/html` and didn't include JSON in the accepted formats. Seems like a shame to me...

Shouldn't this setting rather be a default? Like:

```
if ( <accepts json> ) {
    return json
}
else if ( <accepts html> ) {
    return html
}
else {
    return according to hint
}
```

As far as I understand, the use case was when the client was using some custom `application/something+json`, in which case we will end up in the third branch if this if/else, as intended.

@cescoffier I'll try to merge this PR with mine (#20230) so that we do exactly that. If that's fine for everyone?